File: RetrofitDemo/app/src/main/java/com/example/demo/retrofit/Call.java
package com.example.demo.retrofit;

/**
 * @author tengfei
 * date 2019-12-01 21:40
 * email arrayadapter.cn@gmail.com
 * description
 */
public interface Call<T> {

    void enqueue(CallBack<T> callBack);
}

File: libraryStudy/kotlindemo/src/main/java/com/amazing/tengfei/kotlindemo/utils/Call.kt
package com.amazing.tengfei.kotlindemo.utils

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import kotlin.coroutines.suspendCoroutine

/**
 * @author 飞一般的感觉
 * date 2020/5/17 3:07 PM
 * email arrayadapter.cn@gmail.com
 * description
 */

suspend fun <T> Call<T>.await(): T {

    Result
    return suspendCoroutine {
        enqueue(object : Callback<T> {
            override fun onFailure(call: Call<T>, t: Throwable) {
            }

            override fun onResponse(call: Call<T>, response: Response<T>) {
            }
        })
    }
}